### PR TITLE
fix to handle snap back correctly

### PR DESCRIPTION
**Pull request by matsu911**

### CODE REVIEW

File: test/go_test.rb [hunk 1]
```diff
@@ -80,6 +80,28 @@ def test_capture_three_stones
...
+      game.black(4,3)
+      game.black(2,4)
+      game.black(5,4)
+      game.black(3,5)
+      game.black(4,5)
+      game.white(4,2)
+      game.white(5,3)
+      game.white(3,4)
+      game.white(4,4)
+      game.white(2,0)
```
Comment (jphager2) [inline test/go_test.rb]: What's this about?

File: lib/ruby-go/game.rb [hunk 2]
```diff
@@ -104,17 +104,16 @@ def capture
       stone = @moves.last[:stone]
       stones_around = @board.around(stone)
 
-      captures = stones_around.select {|stn| @board.liberties(stn) == 0}
-
-      captures.each {|stone| capture_group(stone)}
-    end
+      captures = stones_around.
+                   reject {|stn| stn.color == stone.color}.
```
Comment (jphager2) [inline lib/ruby-go/game.rb]: Could you please start the line with the dot?

```ruby
stones_around
  .reject
  .select
```

**Review by jphager2 — COMMENTED** [review]
Hi! Thanks so much for your pull request. It looks very good to me. I really appreciate that you even took the time to make formatting changes. I just have a few comments, so please check them out and I will be happy to accept this pull request. Thanks again.

File: test/go_test.rb (hunk 1 above)
Comment (matsu911) [reply test/go_test.rb]: This test case is the same as below:
https://ja.wikipedia.org/wiki/%E3%82%A6%E3%83%83%E3%83%86%E3%82%AC%E3%82%A8%E3%82%B7

Comment (jphager2) [reply test/go_test.rb]: Yeah I'm just wondering about the move at
White(0, 2)

Comment (matsu911) [reply test/go_test.rb]: Just a mistake.
Removed.

**Pull request merged**